Proofs/Experiments/IsabelleHard.lean

```diff
@@ -0,0 +1,238 @@
+/-
+Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author(s): Siddharth Bhat
+
+This file contains bitvector theorems that stress Isabelle automation.
+We port the file to Lean for comparison.
+-/
+
+import LeanSAT
+open BitVec
+
+/--
+`mask w n` is an w-bit word where the low `n`
+bits are `1`s and the upper bits are `0`s.
+ -/
+def mask (w : Nat) (i : Nat) : BitVec w :=
+  match i with
+  | 0 => 0#w
+  | i' + 1 => 1#w <<< i' ||| mask w i'
```

Comment:
Rather than a recursive match, can you just make this left shift of allOnes and then negate this? This seem to preserver more of the structure.